packages/synapse-react-client/src/utils/functions/EntityTypeUtils.ts

```diff
@@ -306,6 +307,15 @@ export function getVersionDisplay(entity: Entity): string {
   }
 }
 
+export function normalizeSynPrefix(
+  ref: Reference,
+  newDataEntityIds: Set<unknown>,
+) {
+  return !newDataEntityIds.has(
+    ref.targetId.startsWith('syn') ? ref.targetId : `syn${ref.targetId}`,
```

Comment:
Sorry for being unclear, can you just keep this line in the util? The `Set.has` part should stay in the component, since that part is component-specific

Reply:
I want to reuse this in PORTALS-2971 !